modules/local/dada2_quality.nf

```diff
@@ -13,12 +13,61 @@ process DADA2_QUALITY {
     output:
     path "${meta}_qual_stats.pdf"            , emit: pdf
     tuple val(meta), path("*_qual_stats.tsv"), emit: tsv
+    path "versions.yml"                      , emit: versions
     path "*.args.txt"                        , emit: args
+    path "*plotQualityProfile.txt"           , emit: warning
 
     script:
     def args = task.ext.args ?: ''
     """
-    dada_quality.r "${meta}_qual_stats" $args
-    echo 'plotQualityProfile\t$args' > "plotQualityProfile.args.txt"
+    #!/usr/bin/env Rscript
+
+    suppressPackageStartupMessages(library(dada2))
+    suppressPackageStartupMessages(library(ShortRead))
+
+    readfiles <- sort(list.files(".", pattern = ".fastq.gz", full.names = TRUE))
+
+    #make list of number of sequences
+    readfiles_length <- countLines(readfiles) / 4
+
+    #use only the first x files when read number gets above 2147483647, read numbers above that do not fit into an INT and crash the process!
+    if ( sum(readfiles_length) > 2147483647 ) {
+        for (i in length(readfiles):1) {
```

Comment:
Wouldn't it be faster when you use an apply?

Reply:
can I then use `break` (or similar) as below to find the maximum number of files (from first to last)? How would that work?

Reply:
So if I understand correctly, you want to get the highest possible number which is lower than `2147483647` right?
If that is the case, then you don't even need an apply

Reply:
`readfiles[which(sum(readfiles_length[1:i]) <= 2147483647 )]` might already do the trick, replacing L35-L40
(don't pin me on this, could not test it and haven't seen the input data 😅 )

Reply:
Hm, not sure we are speaking about the same thing. I have a list of files, I do a list of sequences counts per file. If the total sequence number across all files is above 2147483647, I want to reduce file by file until I reach a total sequence number across all files that is below 2147483647. I do not know how to to this without a loop or such but I am happy to hear a simpler solution.

Reply:
Thats great, thanks!